Introduce mkosi-sandbox and stop using subuids for image builds (#2956)
systemd/mkosi: mkosi/sandbox/resources/mkosi-sandbox.md

% mkosi-sandbox(1)
%
%

# NAME

mkosi-sandbox — Run commands in a custom sandbox

# SYNOPSIS

`mkosi-sandbox [options…] command [arguments]`

# DESCRIPTION

`mkosi-sandbox` runs the given command in a custom sandbox. The sandbox is configured
by specifying command line options that configure individual parts of the sandbox.

If no command is specified, `mkosi-sandbox` will start `bash` in the sandbox.

> Note that this sandbox is not designed to be a security boundary. It's intended purpose

Reviewer: ```suggestion
Note that this sandbox is not designed to be a security boundary. Its intended purpose
```

is to allow running commands in an isolated environment so they are not affected by the
host system.

# OPTIONS

> `--tmpfs`
> :   Takes a single argument. Mounts a new tmpfs in the sandbox at the given path.

Reviewer: Not our usual style, but makes the sentences on arity unnecessary and therefore the whole man page if not shorter at least less repetitive and also drives home that `--arg=val` is not supported, since in our usualy style all CLI args are given as `--arg=`
```suggestion
`--tmpfs target`
:   Mounts a new tmpfs in the sandbox at the given path.
```

`--dev`

> :   Takes a single argument. Sets up a private /dev in the sandbox at the given path in

Reviewer: ```suggestion
:   Takes a single argument. Sets up a private `/dev` in the sandbox at the given path in
```

>     the sandbox. This private /dev will only contain the basic device nodes required

Reviewer: ```suggestion
    the sandbox. This private `/dev` will only contain the basic device nodes required
```

    for a functioning sandbox (e.g. `/dev/null`) and no actual devices.

`--proc`

> :   Takes a single argument. Mounts /proc from the host at the given path in the sandbox.

Reviewer: ```suggestion
:   Takes a single argument. Mounts `/proc` from the host at the given path in the sandbox.
```

    the current user is mapped to itself in the sandbox. Regardless of whether this option
    is specified or not, the current user will have a full set of ambient capabilities in
    the sandbox. This includes `CAP_SYS_ADMIN` which means that the invoked process in the
    sandbox will be able to do bind mounts and other operations.

    If `mkosi-sandbox` is invoked as the root user, this option won't do anything.

`--suppress-chown`
:   Specifying this option causes all calls to `chown()` or similar system calls to become a
    noop in the sandbox. This is primarily useful when invoking package managers in the
    sandbox which might try to `chown()` files to different users or groups which would fail
    unless `mkosi-sandbox` is invoked by a privileged user.

`--unshare-net`
:   Specifying this option makes `mkosi-sandbox` unshare a network namespace if possible.

`--unshare-ipc`
:   Specifying this option makes `mkosi-sandbox` unshare an IPC namespace if possible.

`--version`
:   Show package version.

`--help`, `-h`
:   Show brief usage information.

# EXAMPLES

Start `bash` in the current working directory in its own network namespace as the current user.

```sh

> mkosi-sandbox --bind / / --chdir $PWD --unshare-net

Reviewer: ```suggestion
mkosi-sandbox --bind / / --chdir "$PWD" --unshare-net
```
but probably better
```suggestion
mkosi-sandbox --bind / / --same-dir --unshare-net
```
